internal/worker/remoterelations/relationunitsworker.go

```diff
@@ -27,59 +28,130 @@ type RelationUnitChangeEvent struct {
 	params.RemoteRelationChangeEvent
 }
 
+// Mode indicates whether the relation units worker is for a local or remote
+// model.
+type Mode string
+
+const (
+	// ModeLocal indicates that the worker is for a local model.
+	ModeLocal Mode = "local"
+	// ModeRemote indicates that the worker is for a remote model.
+	ModeRemote Mode = "remote"
+)
+
 // relationUnitsWorker uses instances of watcher.RelationUnitsWatcher to
 // listen to changes to relation settings in a model, local or remote.
 // Local changes are exported to the remote model.
 // Remote changes are consumed by the local model.
 type relationUnitsWorker struct {
 	catacomb catacomb.Catacomb
 
-	mu sync.Mutex
+	mode Mode
 
 	// mostRecentChange is stored here for the engine report.
+	mu               sync.Mutex
 	mostRecentChange RelationUnitChangeEvent
 	changeSince      time.Time
 
 	relationTag names.RelationTag
-	rrw         watcher.RemoteRelationWatcher
+	watcher     watcher.RemoteRelationWatcher
 	changes     chan<- RelationUnitChangeEvent
-	macaroon    *macaroon.Macaroon
-	mode        string // mode is local or remote.
 
+	mapEvent func(RelationUnitChangeEvent) RelationUnitChangeEvent
+
+	clock  clock.Clock
 	logger logger.Logger
 }
 
-func newRelationUnitsWorker(
+func newLocalRelationUnitsWorker(
+	ctx context.Context,
+	facade RemoteRelationsFacade,
 	relationTag names.RelationTag,
```

Comment:
I'd love to further diverge these workers. There's no special need, now that we don't have facades for local and remote, for the local version to use a tag. The change channel it receives doesn't have to be the same as for the remote.